src/components/app/ServiceWorkerManager.js

```diff
@@ -0,0 +1,152 @@
+|};
+
+class ServiceWorkerManager extends PureComponent<Props, State> {
+  state = {
+    installStatus: 'idle',
+    closeNotice: true,
+  };
+
+  _installServiceWorker() {
+    // We use '@mstange/offline-plugin/runtime' here instead of
+    // 'offline-plugin/runtime' because the fork contains the fix from
+    // https://github.com/NekR/offline-plugin/pull/410
+    const runtime = require('@mstange/offline-plugin/runtime');
+    runtime.install({
+      onInstalled: () => {
+        console.log('[ServiceWorker] App is ready for offline usage!');
+      },
+      onUpdating: () => {
+        console.log(
+          '[ServiceWorker] An update has been found and the browser is downloading the new assets.'
+        );
+      },
+      onUpdateReady: () => {
+        console.log(
+          '[ServiceWorker] We have downloaded the new assets and we are ready to go.'
+        );
+        runtime.applyUpdate();
+        this.setState({ installStatus: 'pending' });
+      },
+      onUpdated: () => {
+        console.log(
+          '[ServiceWorker] The new version of the application has been enabled.'
+        );
+        this.setState({ installStatus: 'ready', closeNotice: false });
+      },
+      onUpdateFailed: () => {
+        console.log(
+          '[ServiceWorker] We failed to update the application for an unknown reason.'
+        );
+      },
+    });
+  }
+
+  componentWillMount() {
+    if (process.env.NODE_ENV === 'production') {
+      this._installServiceWorker();
+    }
+  }
+
+  componentDidUpdate() {
+    const { phase } = this.props;
+    const { installStatus } = this.state;
+
+    if (phase !== 'FATAL_ERROR') {
+      return;
+    }
+
+    if (installStatus === 'ready') {
+      this.reloadPage();
+    }
+  }
+
+  reloadPage() {
+    window.location.reload();
+  }
+
+  _onCloseNotice = () => {
+    this.setState({ closeNotice: true });
+  };
+
+  render() {
+    const { dataSource } = this.props;
+    const { installStatus, closeNotice } = this.state;
+
+    if (dataSource !== 'none' && dataSource !== 'public') {
+      return null;
+    }
+
+    if (installStatus !== 'ready') {
+      return null;
+    }
+
+    if (closeNotice) {
+      return null;
+    }
+
+    return (
+      <div className="serviceworker-ready-notice-wrapper">
+        {/* We use the wrapper to horizontally center the notice */}
+        <div className="serviceworker-ready-notice">
+          <p>
+            A new version of the application has been downloaded and is ready to
+            use.
+          </p>
+          <button
+            className="photon-button serviceworker-ready-notice-action-button"
+            type="button"
+            onClick={this.reloadPage}
+          >
+            Reload the application
+          </button>
+          <button
+            aria-label="Hide the reload notice"
+            title="Hide the reload notice"
+            className="photon-button photon-button-ghost serviceworker-ready-notice-close-button"
```

Comment:
The classes `photon-button` and `photon-button-ghost` come from `/res/css/photon-components.css` that I created when I styled the sidebar button. I was quite happy to be able to reuse them :-)